src/main/kotlin/no/nav/syfo/personstatus/infrastructure/ArbeidsuforhetvurderingClient.kt

```diff
@@ -0,0 +1,87 @@
+package no.nav.syfo.personstatus.infrastructure
+
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.micrometer.core.instrument.Counter
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.client.ClientEnvironment
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.httpClientDefault
+import no.nav.syfo.domain.PersonIdent
+import no.nav.syfo.metric.METRICS_NS
+import no.nav.syfo.metric.METRICS_REGISTRY
+import no.nav.syfo.personstatus.application.arbeidsuforhet.ArbeidsuforhetvurderingDTO
+import no.nav.syfo.personstatus.application.arbeidsuforhet.IArbeidsuforhetvurderingClient
+import no.nav.syfo.util.NAV_CALL_ID_HEADER
+import no.nav.syfo.util.NAV_PERSONIDENT_HEADER
+import no.nav.syfo.util.bearerHeader
+import no.nav.syfo.util.callIdArgument
+import org.slf4j.LoggerFactory
+
+class ArbeidsuforhetvurderingClient(
+    private val azureAdClient: AzureAdClient,
+    private val clientEnvironment: ClientEnvironment,
+) : IArbeidsuforhetvurderingClient {
+
+    private val httpClient = httpClientDefault()
+    private val isarbeidsuforhetUrl = "${clientEnvironment.baseUrl}$ARBEIDSUFORHET_API_PATH"
+
+    override suspend fun getVurdering(
+        callId: String,
+        token: String,
+        personIdent: PersonIdent,
+    ): ArbeidsuforhetvurderingDTO {
+        val oboToken = azureAdClient.getSystemToken(
+            scopeClientId = clientEnvironment.clientId,
+        )?.accessToken ?: throw RuntimeException("Failed to request access to Enhet: Failed to get OBO token")
```

Comment:
Her står det `oboToken`, men bruker `.getSystemToken()`. Jeg tenker det burde være OBO-token vi henter ut, siden det er resultatet av en veileder sin handling? Throw-meldingen burde også oppdateres, kanskje noe sånt `Failed to get OBO-token for arbeidsbeidsuforhet vurdering` eller noe.